src/projects/TodoList/components/CategoryList/index.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import './style.css';
+
+class CategoryList extends React.Component {
+  render () {
+    let categoriesInfo = this.props.categoriesInfo; // <-- category array of objects
+    let eachTaskCategory = this.props.taskCategories.map((categoryId, i) => {
+      let categoryColour = categoriesInfo.find( category => category.id === categoryId )
+      let categoryTooltip = categoryColour.category
+      let categoryDotColour = {
+        backgroundColor: categoryColour.colour
+      };
+      return (
+        <span
+          id={i}
+          className="dot"
+          style={categoryDotColour}
+          data-toggle="tooltip"
+          data-placement="top"
+          title={categoryTooltip}></span>
```

Comment:
If you've got nothing inside the element it can be self-closing

```jsx
<span prop="value">
  Content inside so I have to have a closing tag
</span>
```
```jsx
<span prop="value"></span> // no content inside soo...
<span prop="value" /> // self closing - nice and cleeeean
```

Reply:
done